src/Components/Modal.tsx

```diff
@@ -0,0 +1,94 @@
+import { useQuizContext } from "../Context/QuizProvider";
+import { RESET } from "../Reducer/typeValues";
+import { createTimer } from "../Utils/timer";
+
+export const Modal = ({
+  setShowModal,
+  restart,
+  setShowAnswer,
+}: {
+  setShowModal: React.Dispatch<React.SetStateAction<boolean>>;
+  setShowAnswer: React.Dispatch<React.SetStateAction<boolean>>;
+  restart: (
+    newExpiryTimestamp: number,
+    autoStart?: boolean | undefined
+  ) => void;
+}) => {
```

Comment:
Could this type be imported from a separate file, since it's big, readability decreases